File: 程式設計：資料結構答案檔/C++/204.cpp
#include <iostream>
#include <queue>
using namespace std;

int main() {
    int n, x;
    cin >> n >> x;

    queue<int> allIds;
    for (int i = 1; i <= n; i++) {
        allIds.push(i);
    }

    while (allIds.size() > 1) {
        for (int i = 1; i < x; i++) {
            int t = allIds.front();
            allIds.pop();
            allIds.push(t);
        }
        allIds.pop();
    }

    cout << allIds.front() << endl;
    return 0;
}

File: 程式設計：資料結構答案檔/Python/204.py
import queue

n = int(input())
x = int(input())

allIds = queue.Queue()
for i in range(1, n+1):
    allIds.put(i)

while allIds.qsize() > 1:
    for i in range(1, x):
        t = allIds.get()
        allIds.put(t)
    allIds.get()

print(allIds.get())
